pcdsdevices/dccm.py

```diff
@@ -0,0 +1,289 @@
+        PVs to write to. If omitted, we can guess this from the
+        prefix.
+    acr_status_sufix : str
+        Prefix to the SIOC PV that ACR uses to report the move status.
+        For HXR this usually is 'AO805'.
+    """
+    acr_energy = FCpt(BeamEnergyRequest, '{hutch}',
+                      pv_index='{pv_index}',
+                      acr_status_suffix='{acr_status_suffix}',
+                      add_prefix=('suffix', 'write_pv', 'pv_index',
+                                  'acr_status_suffix'),
+                      kind='normal',
+                      doc='Requests ACR to move the energy.')
+
+    def __init__(
+        self,
+        prefix: str,
+        hutch: typing.Optional[str] = None,
+        acr_status_suffix='AO805',
+        pv_index=2,
+        **kwargs
+    ):
+        self.acr_status_suffix = acr_status_suffix
+        self.pv_index = pv_index
+        super().__init__(prefix, **kwargs)
+
+
+class DCCM(BaseInterface, GroupDevice):
+    """
+    The full DCCM assembly.
+
+    Double Channel Cut Monochrometer controlled with a Beckhoff PLC.
+    This includes five axes in total:
+        - 2 for crystal manipulation (TH1/Upstream and TH2/Downstream)
+        - 1 for chamber translation in x direction (TX)
+    - 2 for YAG diagnostics (TXD and TYD)
+
+    Parameters
+    ----------
+    prefix : str
+        Base PV for DCCM motors
+    name : str, keyword-only
+        name to use in bluesky
+    """
+
+    tab_component_names = True
+
+    th1 = Cpt(BeckhoffAxis, ":MMS:TH1", doc="Bragg Upstream/TH1 Axis", kind="normal")
+    th2 = Cpt(BeckhoffAxis, ":MMS:TH2", doc="Bragg Downstream/TH2 Axis", kind="normal")
+    tx = Cpt(BeckhoffAxis, ":MMS:TX", doc="Translation X Axis", kind="normal")
+    txd = Cpt(BeckhoffAxis, ":MMS:TXD", doc="YAG Diagnostic X Axis", kind="normal")
+    tyd = Cpt(BeckhoffAxis, ":MMS:TYD", doc="YAG Diagnostic Y Axis", kind="normal")
+
+    energy = Cpt(
+        DCCMEnergy, '', kind='hinted',
+        doc=(
+            'PseudoPositioner that moves the theta motors in '
+            'terms of the calculated DCCM energy.'
+        ),
+    )
+
+    energy_with_vernier = Cpt(
+        DCCMEnergyWithVernier, '', kind='normal',
+        doc=(
+            'PseudoPositioner that moves the theta motor in '
+            'terms of the calculated DCCM energy while '
+            'also requesting a vernier move.'
+        ),
+    )
+    energy_with_acr_status = FCpt(
+        DCCMEnergyWithACRStatus, '{prefix}', kind='normal',
+        acr_status_suffix='{acr_status_suffix}',
+        add_prefix=('suffix', 'write_pv', 'acr_status_suffix'),
+        doc=(
+            'PseudoPositioner that moves the alio in '
+            'terms of the calculated CCM energy while '
+            'also requesting an energy change to ACR. '
+            'This will wait on ACR to complete the move.'
+        ),
+    )
+
+    def __init__(
+        self,
+        prefix: str = "SP1L0:DCCM",
```

Comment:
Nit: This is a bit overly specific for my taste, but we don't have that many DCCM's to worry about